src/Coordination/KeeperDispatcher.h

```diff
@@ -65,6 +70,10 @@ class KeeperDispatcher
     ThreadFromGlobalPool session_cleaner_thread;
     /// Dumping new snapshots to disk
     ThreadFromGlobalPool snapshot_thread;
+#if USE_AWS_S3
```

Comment:
Overall, `#if USE_AWS_S3` spreaded over code is ugly. I'd add separate structure for aws s3 handling in separate file. So, #ifdef would be used only in couple places. Imagine we'd like to support object storage in GCP and Azure as well (most probably we are).

Reply:
I agree, but I tried to keep things that are similar close, e.g. all queues and all threads + the places where they are handled.
I can try to extract it differently.